model/SearchEngine/Driver/Elasticsearch/ElasticSearchConfig.php

```diff
@@ -44,7 +44,35 @@ public function __construct(ServiceOptionsInterface $serviceOptions)
 
     public function getHosts(): array
     {
-        return $this->serviceOptions->get(self::class, self::OPTION_HOSTS);
+        $hosts = [];
+        foreach ($this->serviceOptions->get(self::class, self::OPTION_HOSTS) as $host) {
+            if (is_array($host) && isset($host['host'])){
+                $hosts[] = ($host['scheme'] ?? 'http') . '://' . $host['host'] . ':' . ($host['port'] ?? 9200);
+            } else {
+                $hosts[] = $host;
+            }
+        }
+        return $hosts;
+    }
+
+    public function getUsername(): ?string
+    {
+        foreach ($this->serviceOptions->get(self::class, self::OPTION_HOSTS) as $host) {
+            if (is_array($host) && isset($host['user'])){
+                return $host['user'];
+            }
+        }
+        return '';
```

Comment:
Following this comment https://github.com/oat-sa/extension-tao-advanced-search/pull/121/files#r1652061941, you can just 

```suggestion
        return $this->getFirstHost()['user'] ?? null;
```